src/Interpreter.lhs

```diff
@@ -129,7 +130,13 @@
     {----------------------------------------------------------------------}
     
 >   eval :: GlEnv -> Expr -> Either String Expr
->   eval env e = evalState (runErrorT (eval' e)) env
+>   eval env e = evalState (runExceptT (eval' e)) env
+
+>   evalDef :: GlEnv -> Definition -> Either String GlEnv
+>   evalDef env (Def var e) = do
+>       ty <- infer e (toTyEnv env)
+>       let gldef = GlDef {glExpr = e, glType = ty}
+>       return $ (var, gldef) : env
```

Comment:
I am wondering if there is a better name for this function than `evalDef`, given that it doesn't seem to be doing any evaluation. How about something like `addDef`? Indeed, that already exists in https://github.com/mbg/simply-typed-ext/blob/3b91d874260a87169a43265489e17af3ba532e67/app/Main.lhs#L35